src/util/AbstractMemory/CalculationUtil.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (July of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+
+#include "util/ConstexprUtils.h"
+
+// Just the number of bytes per memory unit.
+constexpr size_t numberOfBytesPerKB = ad_utility::pow<size_t>(2, 10);
+constexpr size_t numberOfBytesPerMB = ad_utility::pow<size_t>(2, 20);
+constexpr size_t numberOfBytesPerGB = ad_utility::pow<size_t>(2, 30);
+constexpr size_t numberOfBytesPerTB = ad_utility::pow<size_t>(2, 40);
+constexpr size_t numberOfBytesPerPB = ad_utility::pow<size_t>(2, 50);
+
+/*
+@brief Calculate the amount of bytes for a given amount of untis and a given
+amount of bytes per unit.
+
+@return The amount of bytes. Rounded up, if needed.
+*/
+template <typename T>
+requires std::integral<T> || std::floating_point<T>
+size_t constexpr convertMemoryUnitsToBytes(const T& amountOfUnits,
+                                           const size_t& numberOfBytesPerUnit) {
```

Comment:
Pass the arguments by value, they are numbers!

Reply:
Done, see [ad2d852](https://github.com/ad-freiburg/qlever/pull/1032/commits/ad2d85271961b4fb5886f5ed417bc0e87c4398f2).